text/0000-export-report.md

```diff
@@ -0,0 +1,76 @@
+
+Without this information, a platform will need to resort to suboptimal mechanisms such as parsing the exported output or reading the most recently tagged image on the registry. 
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The lifecycle exporter/creator binary will accept an optional flag `report` which provides a path to `report.toml`. When the flag is supplied the export stage will write a toml representation of the built image for platform consumption.
+
+Here is an overview of the suggested schema for report.toml:
+
+```
+[image]
+tags = ["index.docker.io/image/name:latest", "index.docker.io/image/name:other-tag"]
+identifier = "sha256:c8be1b8f4d60d99c281fc2db75e0f56df42a83ad2f0b091621ce19357e19d853"
+```
+
+By representing this in toml, we can extend this information with additional metadata when needed. 
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The export stage will write report.toml after exporting and caching is complete. There is no default path for report.toml as its location will be platform-specific. If the export steps fails to complete no report.toml is expected to be written.
+
+This addition will require a bump in the platform api.
+
+Docker daemon based platforms such as `pack` can read the report.toml by utilizing [docker cp](https://docs.docker.com/engine/reference/commandline/cp/).
+
+Kubernetes based platforms can utilize the [pod termination message](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message) to retrieve the reports on build completion.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This will require an additional flag which might complicate the lifecycle interface.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Not doing anything and continuing to require platforms to retrieve this information via alternative means.
+- Writing only the built image identifier to an unstructured file. This would be simpiler for some platforms to parse but, would limit the options for extensibility.
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [The kbld tool parses the pack output with a regex to find the built image digest](https://github.com/k14s/kbld/blob/5597786d8369e966f2e7217b24fd058f4a910675/pkg/kbld/image/pack.go#L17). 
+- Tekton allows tasks to surface a built image digest with an [OCI image layout index.json](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#surfacing-the-image-digest-built-in-a-task).
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- How/if `pack` should expose this information to users in a programmatic way? 
+- Is there any additional metadata that should be provided in report.toml?
+- Is report the best name for this file and should it be marshaled into toml?
```

Comment:
`export.toml`?

If this is an export report, could it include the layer usage/size/reuse(?) without needing to dig into the docker image itself.

Reply:
Given we have an `analyzed.toml`, maybe `exported.toml` is the most consistent

Reply:
I like `exported.toml`. Updated the rfc.